File: Blind150/2.Two Pointer/GO/two_sum_2.go
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

func twoSum(numbers []int, target int) []int {

    l, r := 0, len(numbers) -1

    for l < r {

        curSum := numbers[l] + numbers[r]

        if curSum > target{

            r = r - 1
        }else if curSum < target{

            l = l + 1
        }else{

            return []int{l+1, r+1}
        }

    }

    return []int{l+1, r+1}

}

File: Blind150/2.Two Pointer/Python/two_sum_2.py
# https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:


        l, r = 0 , len(numbers)-1


        while l < r:

            cur_sum = numbers[l] + numbers[r]

            if cur_sum > target:

                r -= 1

            elif cur_sum < target:

                l += 1

            else:
                return [l+1, r+1]

        return [l+1, r+1] # just to stop the language server complaints
